docker/centos7/Dockerfile

```diff
@@ -276,25 +276,29 @@ RUN if [ "$(uname -m)" == "aarch64" ]; then \
 # The perl one-liner tells the googlebenchmark build to download a specific SHA of googletest.  By default, they track
 # (tracked?) master, which no longer exists, and risks version skew / incompatibility.
 
-# We disable regexp support because the system version of that library only works with GCC.
+# We disable regexp support for Clang because the system version of that library only works with GCC.
 RUN source /opt/rh/devtoolset-${DEVTOOLSET_VERSION}/enable && \
-    cd /opt && \
+    cd /tmp && \
     git clone https://github.com/google/benchmark.git googlebenchmark-f91b6b && \
     cd googlebenchmark-f91b6b/ && \
     git checkout f91b6b42b1b9854772a90ae9501464a161707d1e && \
     perl -p -i -e s/master/2fe3bd994b3189899d93f1d5a881e725e046fdc2/ cmake/GoogleTest.cmake.in && \
-    cmake -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXX_FLAGS="-stdlib=libc++" -DCMAKE_BUILD_TYPE=Release -DBENCHMARK_DOWNLOAD_DEPENDENCIES=on -DBENCHMARK_ENABLE_LTO=true -DRUN_HAVE_STD_REGEX=0 -DRUN_HAVE_POSIX_REGEX=0 && \
-    make -j 16
-
-# Build a g++ version of google benchmark.
-RUN source /opt/rh/devtoolset-${DEVTOOLSET_VERSION}/enable && \
-    cd /opt && \
-    git clone https://github.com/google/benchmark.git googlebenchmark-f91b6b-g++ && \
-    cd googlebenchmark-f91b6b-g++/ && \
-    git checkout f91b6b42b1b9854772a90ae9501464a161707d1e && \
-    perl -p -i -e s/master/2fe3bd994b3189899d93f1d5a881e725e046fdc2/ cmake/GoogleTest.cmake.in && \
-    cmake -DCMAKE_BUILD_TYPE=Release -DBENCHMARK_DOWNLOAD_DEPENDENCIES=on -DBENCHMARK_ENABLE_LTO=true -DRUN_HAVE_STD_REGEX=0 -DRUN_HAVE_POSIX_REGEX=0 && \
-    make -j 16
+    mkdir build-clang && \
+    cd build-clang && \
+    cmake .. -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXX_FLAGS="-stdlib=libc++" -DCMAKE_BUILD_TYPE=Release -DBENCHMARK_DOWNLOAD_DEPENDENCIES=on -DBENCHMARK_ENABLE_LTO=true -DRUN_HAVE_STD_REGEX=0 -DRUN_HAVE_POSIX_REGEX=0 -DCMAKE_INSTALL_PREFIX=/opt/googlebenchmark-clang && \
+    make -j 16 && \
+    mkdir -p /opt/googlebenchmark-clang && \
+    make install && \
+    cd .. && \
+    mkdir build-gcc && \
```

Comment:
lets keep each google benchmark build as a separate `RUN` directive for the sake of clarity. It will also make it easier to cleanup when/if we drop support for `g++`